project-management/src/main/java/life/qbic/projectmanagement/domain/model/experiment/ExperimentalDesign.java

```
@@ -109,52 +111,179 @@ private void loadCollections() {
+          "No variable with name " + oldName + " exists");
+    }
+    if (variableWithName(newName).isPresent()) {
+      throw new ExperimentalVariableExistsException(
+          "Variable with the name " + newName + " already exists.");
+    }
+    try {
+      renameVariableInGroups(oldName, newName);
+    } catch (RuntimeException e) {
+      //rename back to the original name
+      renameVariableInGroups(newName, oldName);
+      throw e;
+    }
+    //only when the variable was renamed in all groups successfully
+    try {
+      renameVariable(oldName, newName);
+    } catch (RuntimeException e) {
+      //rename back to the original name in groups if renaming failed
+      renameVariableInGroups(newName, oldName);
+      throw e;
+    }
+  }
+
+
+  private void renameVariable(String from, String to) {
+    for (ExperimentalVariable experimentalVariable : variables) {
+      if (experimentalVariable.name().value().equals(from)) {
+        experimentalVariable.renameTo(to);
+        return;
+      }
+    }
+    throw new UnknownExperimentalVariableException("No variable with name " + from + " exists");
+  }
+
+  /**
+   * Renames a variable in the condition of this group. If the variable is not defined, does nothing.
+   * @see Condition#renameVariable(String, String)
+   * @param oldName the old name of the variable
+   * @param newName the new name of the variable after rename. Must not be null or blank.
    */
-  @Deprecated(since = "1.10.0", forRemoval = true)
-  Result<VariableLevel, Exception> addLevelToVariable(String variableName,
-      ExperimentalValue level) {
-    Optional<ExperimentalVariable> experimentalVariableOptional = variableWithName(variableName);
-    if (experimentalVariableOptional.isEmpty()) {
-      return Result.fromError(
-          new ExperimentalVariableNotDefinedException(
-              "There is no variable with name " + variableName));
+  private void renameVariableInGroups(String oldName, String newName) {
```

missing null check for old name => fail early

---

also missing exception signature for the exceptions that can be thrown